### PR TITLE
Add PHP 8 as additional requirement

### PR DESCRIPTION
**Pull request by LIQRGV**
Some library that use this package can't get their CI passed since this package restrict them to use PHP 8.
Some example is here
https://github.com/domnikl/statsd-php/pull/77

### CODE REVIEW

File: appveyor.yml
Comment (LIQRGV) [inline appveyor.yml:11]: Windows doesn't support version below 8.0, 
so...here we are.

**Comment by LIQRGV** [comment]
I saw other PR similar with this on #19. 
But mine has some other changes with appveyor / travis.
Can you take a look on this @webmozart, thank you...

File: .travis.yml
Comment (Ocramius) [inline .travis.yml:16]: This can now be `8.0`

Comment (Ocramius) [reply .travis.yml:16]: `allow_failures` should not contain `8.0` anymore

File: appveyor.yml
Comment (Ocramius) [inline appveyor.yml:17]: We need an entry for `8.0`

File: composer.json
Comment (Ocramius) [inline composer.json]: `5.x` doesn't seem realistic/sensible anymore, but that's up to @webmozart to decide

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Review by jakemercurio — APPROVED** [review]

File: composer.json
Comment (Nek-) [inline composer.json]: It's not because PHPUnit is not compatible with old codebases that the package should abandon PHP 5.x. Of course, it seems to be useless, but some people may have issues with the package if this change is not done on a feature version besides all the content of this PR is actually suppose to go to a minor release. (speaking for PHP 5.6 users while I'm not part of this team for a long time...)

It's @webmozart to decide.

Comment (LIQRGV) [reply composer.json]: How about PHP 5.x user stay on old version ?
I don't think we well have much new feature on the future.
This PR intendend only to support PHP 8 onward

Comment (Ocramius) [reply composer.json]: Indeed: old releases will work just fine 👍

**Comment by chriskapp** [comment]
`vimeo/psalm` uses also this dependency, so it would be great if we can merge this PR and tag a new release.

```
  Problem 1
    - vimeo/psalm[3.14.0, ..., 3.18.2] require webmozart/glob ^4.1 -> satisfiable by webmozart/glob[4.1.0].
    - webmozart/glob 4.1.0 requires php ^5.3.3|^7.0 -> your php version (8.0.0) does not satisfy that requirement.
```

**Comment by LIQRGV** [comment]
]

> `vimeo/psalm` uses also this dependency, so it would be great if we can merge this PR and tag a new release.
> 
> ```
>   Problem 1
>     - vimeo/psalm[3.14.0, ..., 3.18.2] require webmozart/glob ^4.1 -> satisfiable by webmozart/glob[4.1.0].
>     - webmozart/glob 4.1.0 requires php ^5.3.3|^7.0 -> your php version (8.0.0) does not satisfy that requirement.
> ```

Actually, I'm updating this package because I'm using `vimeo/psalm` too @chriskapp 
lol

File: composer.json
Comment (localheinz) [inline composer.json]: How about not using unbounded version constraints here?

Also see https://github.com/symfony/symfony/pull/36876#issuecomment-631351947.

Comment (Nek-) [reply composer.json]: https://twitter.com/nicolasgrekas/status/1263023258938548225

https://twitter.com/driesvints/status/1263031553623556099

Comment (Ocramius) [reply composer.json]: Got bitten by this yesterday on one of my dep upgrades, and got something ancient because it had `>=7.2` in its constraints.

Let's please not repeat the same mistakes we did with un-bounded dependency ranges in 2012~2013 :-)

Comment (LIQRGV) [reply composer.json]: So, you mean we should add ^8.0.0 too?

Comment (Ocramius) [reply composer.json]: Correcto 👍

**Comment by Nek-** [comment]
The last commit on this project is in 2016.

@webmozart seems not to be maintaining any open source project for a while... Maybe this project will never be updated?!

**Comment by LIQRGV** [comment]
I was thinking to move it to my public repo @Nek-.
Do you think it's good idea ?

File: composer.json
Comment (LIQRGV) [reply composer.json]: Done @ocramius

**Comment by Nek-** [comment]
@LIQRGV shouldn't this move to friendsofphp instead?

**Comment by Nek-** [comment]
cc @lyrixx (context: https://github.com/webmozart/glob/issues/23 )

**Comment by LIQRGV** [comment]
> @LIQRGV shouldn't this move to friendsofphp instead?

I'm ok anywhere as long as there is someone who maintain it

**Comment by duxthefux** [comment]
@LIQRGV what's the status here? as help needed somewhere?

**Comment by LIQRGV** [comment]
Hi @duxthefux and @Nek-,
what should I do to add this to friendsofphp repo?

**Comment by Nek-** [comment]
I think we should wait for @webmozart because he used to give rights to projects. Maybe somebody can reach him (I know he's busy right now with important real-life stuff but we really need it) so he gives you access to this repository.

**Comment by martinssipenko** [comment]
I reached out to him via email and got this response:

```
I would be glad if someone took over the lead of the package. I've seen the PRs, but I'm lacking the time to work on them. Would you like to take over?
```

So the question is, where could this project be moved to? Are FriendsOfPHP willing to take over (cc @lyrixx), perhaps Psalm (cc @muglug) wants to take over as they are the largest project that uses this

**Comment by Nek-** [comment]
@martinssipenko I think keeping the project here is fine. We just need to elect a maintainer. @LIQRGV seems to be a good candidate WDYT? ^^

_(if nobody wants to do it I can do it)_

**Comment by LIQRGV** [comment]
I'm ok with anything as long as this project updated.
😁

**Comment by MidnightDesign** [comment]
No offense to @LIQRGV, but @Nek- _seems_ a little more experienced with open source. Maybe both, so we don't end up in the same situation again someday. : )

**Comment by martinssipenko** [comment]
I imagine @Ocramius has a ton of stuff on his plate but perhaps he wants to take over? He does have the most experience with OS.

**Comment by Nek-** [comment]
@martinssipenko if you can ask for @LIQRGV and me to have access to the repository it would be fine. We'll try to ensure it works properly in the future. (we need to end this fast instead of useless debate ;) )

**Comment by Ocramius** [comment]
@martinssipenko I don't mind helping, but the biggest problem is usually getting someone to fiddle with the repo configuration, which doesn't work for contributors.

I'd push some release automation tool ASAP, so that we can ease merges, but need access to repo secrets to do so.

Push rights are the smaller part of the task here 😅

**Comment by martinssipenko** [comment]
> @martinssipenko I don't mind helping, but the biggest problem is usually getting someone to fiddle with the repo configuration, which doesn't work for contributors.
> 
> I'd push some release automation tool ASAP, so that we can ease merges, but need access to repo secrets to do so.
> 
> Push rights are the smaller part of the task here 😅

So perhaps transferring/cloning to another account is better approach?

**Comment by Ocramius** [comment]
> So perhaps transferring/cloning to another account is better approach?

Absolutely: it allows multiple owners, which is kinda needed for redundancy.

**Comment by martinssipenko** [comment]
Frankly, I don't know what to do next, reach out to friendsofphp?

**Comment by LIQRGV** [comment]
Transferring to another repo would lead to updating any repo that use this into new repo.
Just saying

**Comment by localheinz** [comment]
@LIQRGV 

> Transferring to another repo would lead to updating any repo that use this into new repo.

Not necessary:

- we could continue to use `webmozart/glob` as package name
- we could use a different package name and replace `webmozart/glob` (while keeping the same namespaces)

**Comment by martinssipenko** [comment]
That is true, but not transferring means things don't change as the original maintainer has no time to fiddle with repo config. If transferred this could marked as abandoned and suggest using a fork.

@Ocramius you wanna fork?

**Comment by duxthefux** [comment]
If we're forking, this would basically mean this repo needs be archived and a replacement suggestions needs to be placed in README and  `composer.json` to make a good dev experience.

**Comment by Nek-** [comment]
If an organization should fork this repo, it should be psalm. This is the project that needs it the most. (5 573 473 downloads for this one and 4 871 588 for psaml)

**Comment by martinssipenko** [comment]
@Nek- I took a closer look, Psalm stopped using this since version 4.1.0.

**Comment by webmozart** [comment]
Hi all! Sorry for being so unresponsive. I'll do my best to help you fix this quickly. Let me consider a few options and then get back to you with proposals.

**Comment by martinssipenko** [comment]
This repo has now been moved to `webmozarts` org and @Ocramius and @Nek- have been made as repository maintainers.

File: .travis.yml
Comment (LIQRGV) [reply .travis.yml:16]: 8.0 added already

**Comment by Ocramius** [comment]
Ported @LIQRGV's commits to #24 - waiting for some administrative mumbo-jumbo to happen before I can start rolling out releases :-)

**Pull request closed without merging**
